services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java

```diff
@@ -313,47 +313,39 @@ protected RealmModel getAdminstrationRealm(RealmManager realmManager) {
     }
 
     /**
-     * Main page of this realm's admin console
-     *
-     * @return
-     * @throws URISyntaxException
+     * Main page of this realm's admin console.
      */
     @GET
     @NoCache
     public Response getMainPage() throws IOException, FreeMarkerException {
-        if (!session.getContext().getUri(UrlType.ADMIN).getRequestUri().getPath().endsWith("/")) {
-            return Response.status(302).location(session.getContext().getUri(UrlType.ADMIN).getRequestUriBuilder().path("/").build()).build();
-        } else {
-            Theme theme = AdminRoot.getTheme(session, realm);
-
-            Map<String, Object> map = new HashMap<>();
-
-            URI adminBaseUri = session.getContext().getUri(UrlType.ADMIN).getBaseUri();
-            String adminBaseUrl = adminBaseUri.toString();
-            if (adminBaseUrl.endsWith("/")) {
-                adminBaseUrl = adminBaseUrl.substring(0, adminBaseUrl.length() - 1);
-            }
-
-            String kcJsRelativeBasePath = adminBaseUri.getPath();
+        final var baseUriInfo = session.getContext().getUri(UrlType.FRONTEND);
+        final var adminUriInfo = session.getContext().getUri(UrlType.ADMIN);
 
-            if(!kcJsRelativeBasePath.endsWith("/")) {
-                kcJsRelativeBasePath = kcJsRelativeBasePath + "/";
-            }
-
-            URI authServerBaseUri = session.getContext().getUri(UrlType.FRONTEND).getBaseUri();
-
-            String authServerBaseUrl = authServerBaseUri.toString();
-            if (authServerBaseUrl.endsWith("/")) {
-                authServerBaseUrl = authServerBaseUrl.substring(0, authServerBaseUrl.length() - 1);
-            }
-
-            map.put("authServerUrl", authServerBaseUrl);
-            // TODO: The 'authUrl' variable is deprecated and only exists to provide backwards compatibility for older themes, it should be removed in a future version.
-            map.put("authUrl", adminBaseUrl);
+        // Redirect to a URL with a trailing slash if the current URL doesn't have one.
+        if (!adminUriInfo.getRequestUri().getPath().endsWith("/")) {
+            return Response.status(302).location(adminUriInfo.getRequestUriBuilder().path("/").build()).build();
+        } else {
+            // Get the base URLs of the server and admin console.
+            final var serverBaseUri = baseUriInfo.getBaseUri();
+            final var adminBaseUri = adminUriInfo.getBaseUri();
+
+            // Strip any trailing slashes from the URLs.
+            final var serverBaseUrl = serverBaseUri.toString().replaceFirst("/+$", "");
+            final var adminBaseUrl = adminBaseUri.toString().replaceFirst("/+$", "");
+
+            final var map = new HashMap<String, Object>();
+            final var theme = AdminRoot.getTheme(session, realm);
+
+            map.put("serverBaseUrl", serverBaseUrl);
```

Comment:
I wonder if we should unify the naming with the server. Instead of `server` use `frontend`? Possibly it would make it more clear to use the [URL Types naming](https://www.keycloak.org/server/hostname#_frontend) in UI impl too.

But it's not blocking.

Reply:
Yeah, I am willing to consider this, the problem is that in the context of the front-end a 'frontendBaseUrl' looks a bit strange. I have no solid preference, I just introduced a new property here to avoid breaking changes to the existing ones.

@stianst WDYT? Now would be the time to lock down any terminology.